### PR TITLE
Implemented GetDataSourceByName API

### PR DESCRIPTION
**Pull request by utkarshcmu**
Resolving #4258

### CODE REVIEW

**Comment by utkarshcmu** [comment]
#4280 failed CircleCI for this PR.

**Comment by utkarshcmu** [comment]
@torkelo  Thanks for fixing #4280 , please trigger another CircleCi build for this PR and review/merge. Thanks. !!!

**Comment by bergquist** [comment]
@utkarshcmu still fails on circle ci. Could you please rebase against master?

**Comment by utkarshcmu** [comment]
@bergquist Rebased and CircleCi passed!!! Please review/merge. Thanks!!!

File: pkg/api/datasources.go [hunk 1]
```diff
@@ -132,3 +132,35 @@ func GetDataSourcePlugins(c *middleware.Context) {
...
+	if err := bus.Dispatch(&query); err != nil {
+		if err == m.ErrDataSourceNotFound {
+			return ApiError(404, "Data source not found", nil)
+		}
+		return ApiError(500, "Failed to query datasources", err)
+	}
+
+	ds := query.Result
+
+	return Json(200, &dtos.DataSource{
```
Comment (torkelo) [inline pkg/api/datasources.go]: The domain model -> dto translation is duplicated twice in this file, can it be refactored out and reused?

Comment (utkarshcmu) [inline pkg/api/datasources.go]: @torkelo Implemented in #4291 :+1:

**Pull request merged**